web/src/app/users/UserContactMethodCreateDialog.js

```diff
@@ -16,69 +18,121 @@ const createMutation = gql`
     }
   }
 `
-
-export default class UserContactMethodCreateDialog extends React.PureComponent {
-  static propTypes = {
-    userID: p.string.isRequired,
-    onClose: p.func,
+const verifyContactMethodMutation = gql`
+  mutation verifyContactMethod($input: VerifyContactMethodInput!) {
+    verifyContactMethod(input: $input)
   }
+`
+export default function UserContactMethodCreateDialog(props) {
+  // values for contact method form
+  const [cmValue, setCmValue] = useState({
+    name: '',
+    type: 'SMS',
+    value: '+1',
+  })
+  // value for verification form
+  const [verValue, setVerValue] = useState({ code: '' })
+  const [sendError, setSendError] = useState('') // error if verification code send fails
+  const [contactMethodID, setContactMethodID] = useState(null) // used for verification mutation
 
-  state = {
-    value: {
-      name: '',
-      type: 'SMS',
-      value: '+1',
-    },
-    errors: [],
+  const onComplete = (data, errors) => {
+    if (!contactMethodID) {
```

Comment:
Instead of these `if` switches at every level, it really feels like we should have 2 separate dialogs, one for verifying a contact method, and one for creating. We can just transition from one to the other on completion.

We can solve for the animation issues (if it looks odd) with props if needed, but I think it would make the flow easier to follow.